test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/SkipSection.java

```diff
@@ -54,22 +163,21 @@ public static SkipSection parse(XContentParser parser) throws IOException {
         }
         String currentFieldName = null;
         XContentParser.Token token;
-        String version = null;
-        String reason = null;
-        List<String> features = new ArrayList<>();
-        List<String> operatingSystems = new ArrayList<>();
+
+        var builder = new SkipSectionBuilder();
+
         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
             } else if (token.isValue()) {
                 if ("version".equals(currentFieldName)) {
-                    version = parser.text();
+                    builder.withVersion(parser.text());
                 } else if ("reason".equals(currentFieldName)) {
-                    reason = parser.text();
+                    builder.withReason(parser.text());
                 } else if ("features".equals(currentFieldName)) {
```

Comment:
Good question, I'm not sure about that myself.
On the one hand, there will be a strong risk of confusion.
On the other hand, this has been there for a lot of time - people might be used to it. 
With cluster features, we will need to probably add a couple of new fields, like `cluster_has_features` and `cluster_lacks_features` - remember, this will be under `skip`, and we probably want to skip mostly based on the fact that a cluster has not some feature. Wdyt?

Reply:
I wish that would be a `requires` instead of a negative `skip` section. That typically leads to inconsistent semantics, e.g. version (skipped if matching) vs. features (skipped if not matching). But anyways ...

Thinking about `cluster_has_features`, do you think there's use cases where a test should be skipped if a cluster has a feature, but it's expected to be absent? Almost seems like an anti pattern to me, but I'm probably being wishful here.

Considering `features` is already using inverted semantics, it's probably fine to have `cluster_features`. But in that case we better rename `features` to `test_runner_features` or similar I think...

Reply:
I think we can (and probably should) have both `cluster_has_features` and `cluster_lacks_features` (or some better name I'm struggling with).
I have a PR ready (will push them in a moment) with both of them. There are definitely cases for skipping based on features present - it's the mirror of today. You want to split tests so that one runs before a change and one after (e.g. a change in response that needs different assertions). Today this is done with version ranges (e.g. [ - 8.11.99] and [8.12.0 - ]).

Yes, the double negative semantics gives headaches :) I ended up renaming things 2 or 3 times in the next PR.